database/relatedData.js

```diff
@@ -39,6 +38,22 @@ const saveTrack = function (trackData, cb) {
   })
 };
 
+const saveManyTracks = function (tracksData, cb) {
+  // console.log('tracksData');
+  // console.log(tracksData);
+  Track.insertMany(tracksData, (err, tracks) => {
```

Comment:
Model.insertMany also returns a promise, so one suggestion is to refactor with async/await promise syntax instead of callbacks (see my comment below).

Personally, I find async/await easier to use than the callback style.